1st submit

mario-oberwalder opened the pull request:
should work, looking for feedback
kind regards, mario

mario-oberwalder committed:
initial commit

mario-oberwalder committed:
removed unused variables

gerhofer commented on src/main/java/at/codersbay/datastructures/Fractional.java:
```diff
@@ -5,70 +5,133 @@
-        // TODO implement to fix tests in FractionalTest1
+    	/*numerator can be 0 but not null*/
+    	if(numerator == null) {
+    		throw new IllegalArgumentException();
```
IIRC the test requires the exception to have a specific message ;) you sure the tests pass with this?

mario-oberwalder replied:
Tests passed,  didn't mind a missing or false message.
I'm not to firm with the Testing annotation & syntax,
maybe needs an assertEquals for the message following the assertThrows?
https://www.journaldev.com/21826/junit-assert-exception-expected

gerhofer commented on src/main/java/at/codersbay/datastructures/Fractional.java:
```diff
@@ -5,70 +5,133 @@
+	public boolean equals(Object obj) {
+		Fractional fractionalObject;
+		/*!Obj is not a Fractional or is null*/
+		if (!obj.getClass().equals(this.getClass()) || obj == null) {
```
I would rather return false here instead of throwing an IAE - but that's a design decision

mario-oberwalder replied:
aye

gerhofer commented on src/main/java/at/codersbay/datastructures/Fractional.java:
```diff
@@ -5,70 +5,133 @@
+		}
+		/*Numerators are equal after multiplication with other
+		 * fractionals denominator*/
+		if(this.getNumerator()*fractionalObject.getDenominator() 
```
I think we had this before: 
instead of 
```
if (condition) {
  return true; 
}
return false;
```
you can make your life a little easier and write 
`return condition;`

mario-oberwalder replied:
Yes that is something i miss quite often.
This case needs to check for three conditions though.
if (condition) {
 return true;
}
if (condition) {
 return true;
}
if (condition) {
 return true;
}
return false;

I guess i should have chained them with else if() and else().

mario-oberwalder replied:
Yes i tend to miss that, 
in this case since there are three linked if(conditions) i think i will stay with the if(condition) syntax.
I guess i should have chained them with else if(), & else though.

mario-oberwalder replied:
found a way in the new commit

gerhofer commented on src/main/java/at/codersbay/datastructures/Fractional.java:
```diff
@@ -5,70 +5,133 @@
     public Fractional multiply(Fractional other) {
-        // TODO implement to fix tests in FractionalTest4
-        return new Fractional(1, 1);
+    	Integer calcDenom = 0;
```
not sure why you declared these variables with 0 before instead of just filling them with the result of the multiplication - I wouldn't do that because it's just clutter

mario-oberwalder replied:
aye

gerhofer commented on src/main/java/at/codersbay/datastructures/Fractional.java:
```diff
@@ -5,70 +5,133 @@
     public Fractional subtract(Integer other) {
-        // TODO implement to fix tests in FractionalTest7
-        return new Fractional(1, 1);
+        return this.subtract(new Fractional(other,1));
```
really nice especially the reuse of methods in other methods! 👍

mario-oberwalder replied:
thank you

mario-oberwalder committed:
-refactoring especially equals(Object)
-unified variables: factorialObject, newNum, newDenom

mario-oberwalder committed:
-renamed eclipse project to datastructure
-minor change isValidFractiuonal now returns a Boolean
